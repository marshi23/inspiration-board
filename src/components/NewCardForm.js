import React, { Component } from 'react';
import PropTypes from 'prop-types';
// import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]

class NewCardForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      text: "",
      emoji: ""
    };
  }

  resetState = () => {
    this.setState({
      text: "",
      emoji: "",
    })
  }

  onFormChange = (event) => {
    const field = event.target.name;
    const value = event.target.value

    const updatedState = {};
    updatedState[field] = value;
    this.setState(updatedState)
  }

  onNewCardSubmit = (event) => {
    event.preventDefault();

    this.props.addCardCallback(this.state);
    this.resetState();
  }

  render() {

    const emojiOptions = EMOJI_LIST.map((emoji, index) => {
        return <option key={index} value={emoji}>{emoji}</option>
      })

    return (
      <form className="" onSubmit={this.onNewCardSubmit}>
      <header className="form-header">Add Inspiration Card</header>

      <div>
      <label className="" htmlFor="text">Text</label>
      </div>
      <textarea className="" placeholder="Text" name="text" onChange={this.onFormChange} value={this.state.text}></textarea>

      <div>
      <label className="">Emoji</label>
      </div>
      <select name="emoji" onChange={this.onFormChange} className="">
         {emojiOptions}
      </select>

      <section>
        <input className="form-button" type="submit" value="Add Card"/>
      </section>
      </form>
    );
  }
}

NewCardForm.propTypes = {
  addCardCallback: PropTypes.func.isRequired,
};

export default NewCardForm;
