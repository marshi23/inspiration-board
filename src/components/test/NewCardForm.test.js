import React from 'react';
import NewCardForm from '../NewCardForm';
import { shallow } from 'enzyme';


describe('NewCardForm', () => {
  it('test it matches and exisiting snapshot', () => {
    const wrapper = shallow(
      <NewCardForm
        addCardCallback={() => {}}
      />);

    expect(wrapper).toMatchSnapshot();
  });

});
