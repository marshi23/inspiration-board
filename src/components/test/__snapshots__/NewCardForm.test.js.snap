// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm test it matches and exisiting snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewCardForm
    addCardCallback={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="form-header"
        >
          Add Inspiration Card
        </header>,
        <div>
          <label
            className=""
            htmlFor="text"
          >
            Text
          </label>
        </div>,
        <textarea
          className=""
          name="text"
          onChange={[Function]}
          placeholder="Text"
          value=""
        />,
        <div>
          <label
            className=""
          >
            Emoji
          </label>
        </div>,
        <select
          className=""
          name="emoji"
          onChange={[Function]}
        >
          <option
            value=""
          >
            
          </option>
          <option
            value="heart_eyes"
          >
            heart_eyes
          </option>
          <option
            value="beer"
          >
            beer
          </option>
          <option
            value="clap"
          >
            clap
          </option>
          <option
            value="sparkling_heart"
          >
            sparkling_heart
          </option>
          <option
            value="heart_eyes_cat"
          >
            heart_eyes_cat
          </option>
          <option
            value="dog"
          >
            dog
          </option>
        </select>,
        <section>
          <input
            className="form-button"
            type="submit"
            value="Add Card"
          />
        </section>,
      ],
      "className": "",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add Inspiration Card",
          "className": "form-header",
        },
        "ref": null,
        "rendered": "Add Inspiration Card",
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <label
            className=""
            htmlFor="text"
          >
            Text
          </label>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Text",
            "className": "",
            "htmlFor": "text",
          },
          "ref": null,
          "rendered": "Text",
          "type": "label",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "",
          "name": "text",
          "onChange": [Function],
          "placeholder": "Text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <label
            className=""
          >
            Emoji
          </label>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Emoji",
            "className": "",
          },
          "ref": null,
          "rendered": "Emoji",
          "type": "label",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value=""
            >
              
            </option>,
            <option
              value="heart_eyes"
            >
              heart_eyes
            </option>,
            <option
              value="beer"
            >
              beer
            </option>,
            <option
              value="clap"
            >
              clap
            </option>,
            <option
              value="sparkling_heart"
            >
              sparkling_heart
            </option>,
            <option
              value="heart_eyes_cat"
            >
              heart_eyes_cat
            </option>,
            <option
              value="dog"
            >
              dog
            </option>,
          ],
          "className": "",
          "name": "emoji",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "",
              "value": "",
            },
            "ref": null,
            "rendered": "",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "heart_eyes",
              "value": "heart_eyes",
            },
            "ref": null,
            "rendered": "heart_eyes",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": "beer",
              "value": "beer",
            },
            "ref": null,
            "rendered": "beer",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": "clap",
              "value": "clap",
            },
            "ref": null,
            "rendered": "clap",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "children": "sparkling_heart",
              "value": "sparkling_heart",
            },
            "ref": null,
            "rendered": "sparkling_heart",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "host",
            "props": Object {
              "children": "heart_eyes_cat",
              "value": "heart_eyes_cat",
            },
            "ref": null,
            "rendered": "heart_eyes_cat",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "host",
            "props": Object {
              "children": "dog",
              "value": "dog",
            },
            "ref": null,
            "rendered": "dog",
            "type": "option",
          },
        ],
        "type": "select",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="form-button"
            type="submit"
            value="Add Card"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-button",
            "type": "submit",
            "value": "Add Card",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "section",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="form-header"
          >
            Add Inspiration Card
          </header>,
          <div>
            <label
              className=""
              htmlFor="text"
            >
              Text
            </label>
          </div>,
          <textarea
            className=""
            name="text"
            onChange={[Function]}
            placeholder="Text"
            value=""
          />,
          <div>
            <label
              className=""
            >
              Emoji
            </label>
          </div>,
          <select
            className=""
            name="emoji"
            onChange={[Function]}
          >
            <option
              value=""
            >
              
            </option>
            <option
              value="heart_eyes"
            >
              heart_eyes
            </option>
            <option
              value="beer"
            >
              beer
            </option>
            <option
              value="clap"
            >
              clap
            </option>
            <option
              value="sparkling_heart"
            >
              sparkling_heart
            </option>
            <option
              value="heart_eyes_cat"
            >
              heart_eyes_cat
            </option>
            <option
              value="dog"
            >
              dog
            </option>
          </select>,
          <section>
            <input
              className="form-button"
              type="submit"
              value="Add Card"
            />
          </section>,
        ],
        "className": "",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add Inspiration Card",
            "className": "form-header",
          },
          "ref": null,
          "rendered": "Add Inspiration Card",
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <label
              className=""
              htmlFor="text"
            >
              Text
            </label>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Text",
              "className": "",
              "htmlFor": "text",
            },
            "ref": null,
            "rendered": "Text",
            "type": "label",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "",
            "name": "text",
            "onChange": [Function],
            "placeholder": "Text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <label
              className=""
            >
              Emoji
            </label>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Emoji",
              "className": "",
            },
            "ref": null,
            "rendered": "Emoji",
            "type": "label",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value=""
              >
                
              </option>,
              <option
                value="heart_eyes"
              >
                heart_eyes
              </option>,
              <option
                value="beer"
              >
                beer
              </option>,
              <option
                value="clap"
              >
                clap
              </option>,
              <option
                value="sparkling_heart"
              >
                sparkling_heart
              </option>,
              <option
                value="heart_eyes_cat"
              >
                heart_eyes_cat
              </option>,
              <option
                value="dog"
              >
                dog
              </option>,
            ],
            "className": "",
            "name": "emoji",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "",
                "value": "",
              },
              "ref": null,
              "rendered": "",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "heart_eyes",
                "value": "heart_eyes",
              },
              "ref": null,
              "rendered": "heart_eyes",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": "beer",
                "value": "beer",
              },
              "ref": null,
              "rendered": "beer",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "children": "clap",
                "value": "clap",
              },
              "ref": null,
              "rendered": "clap",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "host",
              "props": Object {
                "children": "sparkling_heart",
                "value": "sparkling_heart",
              },
              "ref": null,
              "rendered": "sparkling_heart",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "5",
              "nodeType": "host",
              "props": Object {
                "children": "heart_eyes_cat",
                "value": "heart_eyes_cat",
              },
              "ref": null,
              "rendered": "heart_eyes_cat",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "6",
              "nodeType": "host",
              "props": Object {
                "children": "dog",
                "value": "dog",
              },
              "ref": null,
              "rendered": "dog",
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-button"
              type="submit"
              value="Add Card"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-button",
              "type": "submit",
              "value": "Add Card",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "section",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
